programs/drift/src/instructions/admin.rs

```diff
@@ -625,6 +631,50 @@ pub fn handle_initialize_perp_market(
             } = get_prelaunch_price(&ctx.accounts.oracle, clock_slot)?;
             (oracle_price, oracle_delay, oracle_price)
         }
+        OracleSource::PythPull => {
```

Comment:
nit: can maybe merge the Pyth and PythPull branches and just add a method is_pull to oracle source (if we think less lines of code is more readable

Reply:
i initially did this, but was thinking it would be nice to explicitly define the oracle source like this so we dont have to worry about making sure we correctly set the switch when we call `get_oracle_price` -- it's already inferred whether to use push or pull from the explicitly stated OracleSource